ztp/ztp-ran-policy-generator/README.md

```diff
@@ -1,11 +1,265 @@
+          object-templates:
+            - complianceType: musthave
+              objectDefinition:
+                apiVersion: machineconfiguration.openshift.io/v1
+                kind: MachineConfigPool
+                metadata:
+                  name: worker-du
+                  labels:
+                    machineconfiguration.openshift.io/role: worker-du
+                spec:
+                  machineConfigSelector:
+                    matchExpressions:
+                      - key: machineconfiguration.openshift.io/role
+                        operator: In
+                        values: [worker, worker-du]
+                  nodeSelector:
+                    matchLabels:
+                      node-role.kubernetes.io/worker-du: ""
+```
+
+3 - RanGenTemplate:
+
+RanGenTemplate is a Custom Resource Definition (CRD) that tells RanGen where to locate the generated policies and which Spec items need to be defined. Check the [RanGenTemplate](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ztp-ran-policy-generator/ranGenTemplates/RanGenTemplate.yaml) for more info. Let's consider the [group-du-ranGen.yaml](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ztp-ran-policy-generator/ranGenTemplateExamples/group-du-ranGen.yaml) example to explain more
+
+```
+apiVersion: ranPolicyGenerator/v1
+kind: RanGenTemplate
+metadata:
+ name: "group-du-policies"
+ namespace: "policy-template"
+ labels:
+   common: false
+   groupName: "group-du"
+   siteName: "N/A"
+   mcp: "worker-du"
+sourceFiles:
+ - fileName: MachineConfigPool
+   policyName: "mcp-du-policy"
+   name: "worker-du"
+ - fileName: SriovOperatorConfig
+   policyName: "sriov-operconfig-policy"
+ - fileName: MachineConfigSctp
+   policyName: "mc-sctp-policy"
+ - fileName: MachineConfigContainerMountNS
+   policyName: "mc-mount-ns-policy"
+ - fileName: MachineConfigDisableChronyd
+   policyName: "mc-chronyd-policy"
+ - fileName: PtpConfigSlave
+   policyName: "ptp-config-policy"
+   name: "du-ptp-slave"
+   spec:
+     profile:
+     - name: "slave"
+       interface: "ens5f0"
+       ptp4lOpts: "-2 -s --summary_interval -4"
+       phc2sysOpts: "-a -r -n 24"
+```
+
+The group-du-ranGen.yaml defines a group of policies under a group named group-du. It defines a MachineConfigPool worker-du that will be used as the node selector for any other policy defined under the sourceFiles. The source file PtpConfigSlave has a definition of a [PtpConfig](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ztp-ran-policy-generator/sourcePolicies/PtpConfigSlave.yaml) CR. The generated policy for PtpConfig will be named as group-du-ptp-config-policy. The PtpConfig CR defined in the group-du-ptp-config-policy will be named as du-ptp-slave. The spec defined in the PtpConfigSlave will be placed under du-ptp-slave in the generated policy along with the other spec items defined under the source file. The group-du-ptp-config-policy will be as below
```

Comment:
is the reference to PtpConfig an example - maybe best to highlight is an example of one parameter.

Reply:
yes as an example, okay will make it more clear.